components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -1837,31 +1857,36 @@ public Set<AccessTokenDO> getAccessTokensByUserForOpenidScope(AuthenticatedUser
     }
 
     private Map<String, AccessTokenDO> getAccessTokenDOMapFromResultSet(AuthenticatedUser authenticatedUser,
-                                                                        ResultSet rs) throws SQLException,
-            IdentityOAuth2Exception {
+                ResultSet rs, boolean includeExpiredAccessTokensWithActiveRefreshToken)
+            throws SQLException, IdentityOAuth2Exception {
 
         Map<String, AccessTokenDO> tokenMap = new HashMap<>();
         while (rs.next()) {
             Timestamp timeCreated = rs.getTimestamp(4, Calendar.getInstance(TimeZone.getTimeZone(UTC)));
             long issuedTimeInMillis = timeCreated.getTime();
             long validityPeriodInMillis = rs.getLong(5);
+            Timestamp refreshTokenTimeCreated = rs.getTimestamp(6, Calendar.getInstance(TimeZone.getTimeZone(UTC)));
```

Comment:
do we need to get this information if `includeExpiredAccessTokensWithActiveRefreshToken` is false?

Reply:
if `includeExpiredAccessTokensWithActiveRefreshToken ` is false and if the access token we have retrieved is already expired, in that case we dont require this information at this point.